front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/ItineraryMarkers.tsx

```diff
@@ -65,122 +66,164 @@ const formatPointWithNoName = (
   </>
 );
 
-const extractMarkerInformation = (pathSteps: MarkerInformation[], showStdcmAssets: boolean) =>
-  pathSteps.reduce((acc, cur, index) => {
-    if (cur && cur.coordinates) {
-      if (index === 0) {
-        acc.push({
-          coordinates: cur.coordinates,
-          type: MARKER_TYPE.ORIGIN,
-          marker: cur,
-          imageSource: showStdcmAssets ? stdcmOrigin : originSVG,
-        });
-      } else if (index > 0 && index < pathSteps.length - 1) {
-        acc.push({
-          coordinates: cur.coordinates,
+const extractMarkerInformation = (
+  pathSteps: MarkerInformation[],
+  showStdcmAssets: boolean,
+  suggestedOP: SuggestedOP[]
+): MarkerProperties[] =>
+  pathSteps
+    .map((pathStep, index): MarkerProperties | null => {
+      const matchingOp = suggestedOP.find((op) => matchPathStepAndOp(pathStep, op));
+
+      if (!matchingOp) return null;
+
+      if (pathStep.coordinates) {
+        if (index === 0) {
+          return {
+            coordinates: pathStep.coordinates,
+            type: MARKER_TYPE.ORIGIN,
+            imageSource: showStdcmAssets ? stdcmOrigin : originSVG,
+            op: matchingOp,
+            pathStep,
+          };
+        }
+
+        if (index === pathSteps.length - 1) {
+          return {
+            coordinates: pathStep.coordinates,
+            type: MARKER_TYPE.DESTINATION,
+            imageSource: showStdcmAssets ? stdcmDestination : destinationSVG,
+            op: matchingOp,
+            pathStep,
+          };
+        }
+
+        return {
+          coordinates: pathStep.coordinates,
           type: MARKER_TYPE.VIA,
-          marker: cur,
           imageSource: showStdcmAssets ? stdcmVia : viaSVG,
           index,
-        });
-      } else if (index === pathSteps.length - 1) {
-        acc.push({
-          coordinates: cur.coordinates,
-          type: MARKER_TYPE.DESTINATION,
-          marker: cur,
-          imageSource: showStdcmAssets ? stdcmDestination : destinationSVG,
-        });
+          op: matchingOp,
+          pathStep,
+        };
       }
-    }
-    return acc;
-  }, [] as MarkerProperties[]);
+      return null;
+    })
+    .filter((marker): marker is MarkerProperties => marker !== null);
+
+const ItineraryMarkers = ({
+  map,
+  simulationPathSteps,
+  pathStepsAndSuggestedOPs,
+  showStdcmAssets,
+}: ItineraryMarkersProps) => {
+  const { getInfraID } = useOsrdConfSelectors();
+  const infraId = useSelector(getInfraID);
+  if (!infraId) return;
```

Comment:
A component shouldn't return `undefined`, you can return `null` instead.
If yo also add `if (!markersInformation) return null;` before the component return, you will be able to remove your eslint comment and the `?` in `markersInformation?.map`
```suggestion
  if (!infraId) return null;
```

Reply:
Done